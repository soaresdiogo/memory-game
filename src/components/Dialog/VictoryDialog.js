import React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';

const VictoryDialog = ({ isOpen, onClose, onGameRestart }) => (
  <Dialog data-testid="victoryDialog" open={isOpen} onClose={onClose}>
    <DialogTitle data-testid="victoryDialogTitle">Congrats!</DialogTitle>
    <DialogContent>
      <DialogContentText data-testid="victoryDialogContentText">
        You win the game.
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button
        data-testid="victoryDialogRestartButton"
        color="primary"
        onClick={onGameRestart}
      >
        Restart
      </Button>
      <Button
        data-testid="victoryDialogOkButton"
        color="primary"
        onClick={onClose}
        autoFocus
      >
        Ok
      </Button>
    </DialogActions>
  </Dialog>
);

VictoryDialog.defaultProp = {
  onClose: PropTypes.func,
  onGameRestart: PropTypes.func,
};

VictoryDialog.propTypes = {
  isOpen: PropTypes.bool.isRequired,
};

export default VictoryDialog;
