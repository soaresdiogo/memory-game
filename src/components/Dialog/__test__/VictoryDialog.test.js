import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import VictoryDialog from '../VictoryDialog';

import '@testing-library/jest-dom/extend-expect';

const renderVictoryDialog = () => {
  const { getByTestId } = render(<VictoryDialog isOpen />);
  const container = getByTestId('victoryDialog');
  return {
    getByTestId,
    container,
  };
};

describe('Victory dialog component', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<VictoryDialog isOpen />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('renders all elements correctly', () => {
    const { container, getByTestId } = renderVictoryDialog();
    expect(container.children.length).toBe(4);
    expect(getByTestId('victoryDialogTitle')).toBeDefined();
    expect(getByTestId('victoryDialogContentText')).toBeDefined();
    expect(getByTestId('victoryDialogRestartButton')).toBeDefined();
    expect(getByTestId('victoryDialogOkButton')).toBeDefined();
  });

  test('should title be Congrats!', () => {
    const { getByTestId } = renderVictoryDialog();
    expect(getByTestId('victoryDialogTitle').textContent).toBe('Congrats!');
  });

  test('should content text be You win the game', () => {
    const { getByTestId } = renderVictoryDialog();
    expect(getByTestId('victoryDialogContentText').textContent).toBe(
      'You win the game.',
    );
  });

  test('should be a button with value Restart', () => {
    const { getByTestId } = renderVictoryDialog();
    expect(getByTestId('victoryDialogRestartButton').type).toBe('button');
    expect(getByTestId('victoryDialogRestartButton').textContent).toBe(
      'Restart',
    );
  });

  test('should be a button with value Ok', () => {
    const { getByTestId } = renderVictoryDialog();
    expect(getByTestId('victoryDialogOkButton').type).toBe('button');
    expect(getByTestId('victoryDialogOkButton').textContent).toBe('Ok');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<VictoryDialog isOpen={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
