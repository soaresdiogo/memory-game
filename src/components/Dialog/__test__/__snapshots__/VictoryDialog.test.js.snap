// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Victory dialog component matches snapshot 1`] = `null`;
